CONTRIBUTING.md

```diff
@@ -123,6 +124,96 @@ from the main (upstream) repository:
+we use the git commit messages to **generate changelog**.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of
+the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is
+the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+  semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests or correcting existing tests
+* **build**: Changes that affect the build system, CI configuration or external dependencies
+            (example scopes: gulp, broccoli, npm)
+* **chore**: Other changes that don't modify `src` or `test` files
+
+### Scope
+The scope could be anything specifying place of the commit change. For example
+`core`, `schema`, `query-builder`, `postgres`, `mysql`, `mssql` etc.
+
+### Subject
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* don't capitalize first letter
+* no dot (.) at the end
+
+### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+reference GitHub issues that this commit **Closes**.
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines.
+The rest of the commit message is then used for this.
+
+### Examples
+Fix and close issue:
+```
+fix(mysql): resolve issues uppercase column names
+
+Closes: #123456
+```
+Implement new feature:
+```
+feat: implement new magic decorator
+
+This new feature change bahviour of typeorm to allow use new magic decorator...
+
+Closes: #22222
+```
+Doc update:
+```
+doc(mssql): update supported mssql column types
+```
+Breaking change:
+```
+refactor(core): refactor driver API and generate separate packages per driver
+
+BREAKING CHANGE: it is necessary install new core package `npm install @typeorm/core` and specific driver `npm install @typeorm/postgres`
```

Comment:
I would advice to change this example since we don't have such packages yet. It's going to happen some day, but it's not certain `@typeorm/postgres` will be an exact name.

Reply:
Sure agree :) Will change this.